app/DashBoard/Controls/AddUser/Control.php

```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types = 1);
+
+namespace Pd\Monitoring\DashBoard\Controls\AddUser;
+
+class Control extends \Nette\Application\UI\Control
+{
+
+	private \Pd\Monitoring\DashBoard\Forms\UserAddFormFactory $userAddFormFactory;
+
+	private \Pd\Monitoring\User\UsersRepository $usersRepository;
+
+	private \Nette\Security\User $user;
+
+
+	public function __construct(
+		//\Pd\Monitoring\User\User $identity,
+		\Pd\Monitoring\DashBoard\Forms\UserAddFormFactory $userAddFormFactory,
+		\Pd\Monitoring\User\UsersRepository $usersRepository,
+		\Nette\Security\User $user
+	)
+	{
+		$this->userAddFormFactory = $userAddFormFactory;
+		$this->usersRepository = $usersRepository;
+		$this->user = $user;
+	}
+
+
+	public function render(): void
+	{
+		$this->template->setFile(__DIR__ . '/Control.latte');
+		$this->template->render();
+	}
+
+
+	protected function createComponentForm(): \Nette\Application\UI\Form
+	{
+		$form = $this->userAddFormFactory->create();
+
+
+		$form->onSuccess[] = function (\Nette\Forms\Form $form, \Pd\Monitoring\DashBoard\Forms\UserEditFormData $values): void
+		{
+			$this->processAddForm($form, $values);
+		};
+
+		return $form;
+	}
+
+
+	protected function processAddForm(\Nette\Forms\Form $form, \Pd\Monitoring\DashBoard\Forms\UserEditFormData $values): void
+	{
+
+		$user = new \Pd\Monitoring\User\User();
+		$user->gitHubName = $values->gitHubName;
+		$user->slackId = $values->slackId;
+		$user->administrator = $values->administrator;
+		$user->email = $values->email;
+
+		$passwordGenerator = new \Nette\Security\Passwords();
```

Comment:
Je to služba: https://github.com/peckadesign/Monitoring/blob/master/app/DashBoard/Controls/EditUser/Control.php#L24